api/envoy/api/v2/cds.proto

```diff
@@ -236,6 +236,82 @@ message Cluster {
   //   verification.
   auth.UpstreamTlsContext tls_context = 11;
 
+  // TransportSocketMatch specifies what transport socket config will be used
+  // when the matches conditions are satisfied.
+  message TransportSocketMatch {
+    // The name of the match.
```

Comment:
Should this be required? What does this do?

Reply:
Functionality not required given now we're using key/value matching API. But I do plan to use the name as stats/debugging log for troubleshooting. Open to other suggestions as well.

Reply:
That sounds fine. Can we make it required then with a validation?

Reply:
Done. add a min_len = 1, since we don't have required validation for string type per https://github.com/mwitkow/go-proto-validators.